### PR TITLE
maint: Stop including PF API files for fedora > 33, RHEL > 8 && debian testing

### PR DESCRIPTION
**Pull request by KKoukiou**
This also need to be done for debian - 2 (in adjust-for-release-file?).
Also there is cockpit.css API which is also not supposed to be used. Should we drop it now as well?

### CODE REVIEW

**Comment by martinpitt** [comment]
Thanks! For Debian this can be done in debian/rules. Commit d957152c3 still showed how to do that. So something like this (untested):

```make
# only ship deprecated PatternFly API for stable releases (backports); Debian sid/testing has no VERSION_ID
SHIP_PF_API = $(filter $(shell . /etc/os-release; echo $${VERSION_ID:-unstable}),10 20.04)
```

and later, at the end of `override_dh_install`:
```make
ifeq ($(SHIP_PF_API),)
       rm debian/cockpit-bridge/usr/share/cockpit/base1/patternfly.css
endif
```

File: tools/cockpit.spec [hunk 1]
```diff
@@ -269,6 +269,11 @@ rm -f %{buildroot}/%{_prefix}/share/metainfo/org.cockpit-project.cockpit-machine
 rm -f %{buildroot}/%{_prefix}/share/metainfo/org.cockpit-project.cockpit-storaged.metainfo.xml
 %endif
 
+# Stop including PF API files in fedora > 33 and RHEL > 8
+%if 0%{?fedora} > 33 || 0%{?rhel} > 8
+    rm -r %{_datadir}/cockpit/base1/patternfly.css
```
Comment (martinpitt) [inline tools/cockpit.spec:274]: Please drop the `-r`, it's not a directory.

**Comment by martinpitt** [comment]
test/verify/files/embed-cockpit.html still includes patternfly.css, that should be moved to cockpit.css  for the time being.

cockpit.css is still being used widely, mostly in our examples (where they are bad, hmm) and in pkg/shell/shell.html (where it can hopefully be replaced by a compile-time inclusion of our overrides). We need to talk about what to do with that, but at least cockpit.css isn't an API that we document.

What we certainly *should* do is to drop the patternfly.css bits from cockpit.css, and change the examples to not rely on the PF parts. Then the remaining cockpit.css is hopefully relatively harmless, and we can  look into how to wean off the examples of them.

**Comment by KKoukiou** [comment]
@martinpitt I reposted but I clearly break the UI in some example pages which where using the patternlfy.css file indirectly though . 
If we care about the UI in example pages we can copy in the example folder the patterfly.css file and let them import it from their folder. But honestly these tests are not really important for the UI I feel. What do you think?

Otherwise we can port them to compile their js/css though webpack loaders - like the rest of the pages.

**Comment by martinpitt** [comment]
Tests fail on

    make[1]: *** No rule to make target 'src/base1/cockpit.css', needed by 'distdir-am'.  Stop.

In addition, the debian/rules syntax is wrong. I'll look into that.

**Comment by martinpitt** [comment]
I fixed the Make dependency, debian/rules, and split out the commits. I verified that building ubuntu-stable debs do *not* include patternfly.css any more, but building ubuntu-2004 and fedora-33 *does* contain it still (as intended).

I also confirmed that the examples look PatternFly-ish again with the fixed cockpit.css. But I don't think that is very important, let's try to remove these parts as a next step.

**Comment by martinpitt** [comment]
When I modify the cockpit.spec condition `0%{?fedora} > 32`, then the build fails -- we don't cover this in any supported OS yet, but it will fail on rawhide. Fixed now.

**Comment by martinpitt** [comment]
@marusak: Since I dabbled heavily in this PR now, would you mind doing a final review? Dakujem!

**Review by marusak — APPROVED** [review]
Thanks!
The changes look right, check it out and it does not break anything.

**Pull request merged**